src/scripts/components/lang-redirects.js

```diff
@@ -0,0 +1,100 @@
+import Cookies from 'js-cookie';
+
+const allowedLanguages = ['en', 'ja', 'fr'];
+const enabledSubdomains = [
+	'docs',
+	'docs-staging',
+	'corpsite-preview',
+	'corpsite-staging',
+	'localhost'
+];
+const cookiePath = '/';
+
+const getUrlVars = () => {
+	const vars = {};
+	const href = window.location.href.replace(window.location.hash, '');
+    href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+        vars[key] = value;
+    });
+    return vars;
+}
+
+const getDomain = (data) => {
+    const a = document.createElement('a');
+    a.href = data;
+    return a.hostname;
+}
+
+const getPath = (data) => {
+    const a = document.createElement('a');
+    a.href = data;
+    return a.pathname;
+}
+
+export function handleLanguageBasedRedirects() {
+	const params = getUrlVars();
+
+	const supportedLanguages = navigator.language || navigator.browserLanguage;
+	const baseURL = getDomain(window.location.href);
+	const subdomain = baseURL.split('.')[0];
+	const subMatch = enabledSubdomains.filter((i) => subdomain === i);
+	let uri = getPath(window.location.href);
+	let previewPath = '';
+	let acceptLanguage = "en";
+	let logMsg = '';
+
+
+	/* Update URI based on preview links. Branch/feature needs to be moved in front of language redirect
+		instead of being appended to the end
+		ex: /mybranch/myfeature/index.html => /ja/mybranch/myfeature/index.html
+	*/
+	if ( subdomain.includes('preview') || subdomain.includes('docs-staging') ) {
+		previewPath = uri.split('/').slice(0,3).join('/');
+		uri = uri.replace(preview_path, '');
+	}
+
+	// order of precedence: url > cookie > header
+	if ( params['lang_pref'] && allowedLanguages.indexOf(params['lang_pref']) !== -1 ) {
+		acceptLanguage = params['lang_pref'];
+		logMsg = `Change acceptLanguage based on URL Param: ${ acceptLanguage }`;
+
+		Cookies.set("lang_pref", acceptLanguage, {path: cookiePath});
+	}
+	else if (Cookies.get('lang_pref') && allowedLanguages.indexOf(Cookies.get('lang_pref')) !== -1 ) {
+		acceptLanguage = Cookies.get('lang_pref');
+		logMsg = `Change acceptLanguage based on lang_pref Cookie: ${ acceptLanguage}`;
+	}
+	else if ( supportedLanguages && subMatch.length ) {
+		logMsg = `Set acceptLanguage based on navigator.language header value: ${  supportedLanguages  } ; DEBUG: ${ supportedLanguages.startsWith('ja') }`;
+
+		if ( supportedLanguages.startsWith('ja') ) {
```

Comment:
should we be checking this against `allowedLanguages` instead of just hardcoding to `ja`?

Reply:
Initially, we just want to check for JA because that’s the only redirect we’re supporting at the moment

Reply:
actually, not sure about this.. going to spend some time on it today

Reply:
So yea, I confirmed.. this is the case because we only want to force redirects for JA at the moment but we want to support the other languages. I'll add some comments so this is clear

Reply:
ok sounds good, when i wrote the comment i was mostly thinking about having this defined at the top of the file or somewhere else where we can just add another language if we support fr or something too.

Reply:
yea, good call.. working on doing that now

Reply:
Alright, I've added a separate redirectLanguages where we can set which accept languages should actually trigger a redirect